### ISSUE TITLE
fmi3Callback{ClockUpdate|IntermediateUpdate} should be imperative

### ISSUE THREAD

**Issue by t-sommer** [modelica/fmi-standard#1609]
maybe `UpdateClock(s)` and `UpdateIntermediateVariables`?

**Comment by TorstenBlochwitz** [comment]
Intermediate Update Mode does not only refer to variables. For that reason I would propose to keep the name fmi3CallbackIntermediateUpdate, which is an imperative already (at least kind of).

**Comment by andreas-junghanns** [comment]
Maybe the names are not too bad, considering that the modes these callback lead to are called Clock Update Mode and Intermediate Update Mode.
Please reopen if you have a strong opinion or more arguments.

**Comment by t-sommer** [comment]
Wouldn't `EnterIntermediateUpdateMode` be appropriate? This would follow the "imperative" convention and be consistent with the remaining mode transition functions.

**Comment by andreas-junghanns** [comment]
So it would be `fmi3EnterIntermediateUpdateMode` and `fmi3EnterClockUpdateMode`?
Sounds in many ways more consistent - I like that. @TorstenBlochwitz?
If we do think this is a good change, we should get this done soon, as probably the last change for the next Beta (possibley RC1).

**Comment by t-sommer** [comment]
While we're on it, we should also consider to either move the `Callback` prefix to the end of the names, i.e. `fmi3*Callback` instead of `fmi3Callback*` or better remove it altogether, because it doesn't really convey any information.

**Comment by TorstenBlochwitz** [comment]
fmi3EnterIntermediateUpdateMode and fmi3EnterClockUpdateMode sounds good to me. I checked some through some occurances and it reads OK.

By doing this I noticed a wrong link in 2.2. General Mechanisms: fmi3CallbackIntermediateUpdate points to the capability flag, but should point to the place, the function is explained in 4.2.2. State: Intermediate Update Mode.

Having "callback" at the end might further improve the reading, but I have no strong opinion on this.
